pkg/generate/code/check.go

```diff
@@ -132,37 +136,82 @@ func checkRequiredFieldsMissingFromShape(
 			missing = append(missing, primaryARNCondition)
 			continue
 		}
-		cleanMemberNames := names.New(memberName)
-		cleanMemberName := cleanMemberNames.Camel
 
-		resVarPath := koVarName
-		// Check that the field has potentially been renamed
-		renamedName, wasRenamed := r.InputFieldRename(
-			op.Name, memberName,
-		)
-		_, found := r.SpecFields[renamedName]
-		if found && !wasRenamed {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + cleanMemberName
-		} else if found {
-			resVarPath = resVarPath + r.Config().PrefixConfig.SpecField + "." + renamedName
-		} else {
-			_, found = r.StatusFields[memberName]
-			if !found {
-				// If it isn't in our spec/status fields, we have a problem!
-				msg := fmt.Sprintf(
-					"GENERATION FAILURE! there's a required field %s in "+
-						"Shape %s that isn't in either the CR's Spec or "+
-						"Status structs!",
-					memberName, shape.ShapeName,
-				)
-				panic(msg)
-			}
-			resVarPath = resVarPath + r.Config().PrefixConfig.StatusField + "." + cleanMemberName
+		resVarPath, err := getSanitizedMemberPath(memberName, r, op, koVarName)
+		if err != nil {
+			// If it isn't in our spec/status fields, we have a problem!
+			msg := fmt.Sprintf(
+				"GENERATION FAILURE! there's a required field %s in "+
+					"Shape %s that isn't in either the CR's Spec or "+
+					"Status structs!",
+				memberName, shape.ShapeName,
+			)
+			panic(msg)
 		}
 		missing = append(missing, fmt.Sprintf("%s == nil", resVarPath))
 	}
+
+	// If ReadMany input shape can take a resource identifier or collection of
```

Comment:
Should this code block under condition like if op == r.Ops.ReadMany ??

Reply:
Yeah, was torn between reducing branched logic (constant work) or explicitly separating the logic out per your suggestions. The latter makes it clearer and is consistent with existing code so will move forward with that.

Will break out the PR 👍